### PR TITLE
feat: SDK Logging

### PR DESCRIPTION
**Pull request by yafuquen**
## Jira Ticket
- ACCSEC-19003

## Description
- Add logging to verify and security modules
- Provide a default logger service in TwilioVerify builder, using method enableDefaultLoggingService
- Provide interface and log leves to implement a custom logger service, adding it using addLoggingService in TwilioVerify builder.

## Commit message
-  feat: SDK logging

## Testing
- [ ] Added unit tests
- [ ] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Review by SergioFierro — APPROVED** [review]

**Review by Greshot — APPROVED** [review]

File: TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift [hunk 1]
```diff
@@ -34,11 +34,15 @@ extension URLSession {
...
         return
       }
+      Logger.shared.log(withLevel: .networking, message: "Response code: \(response.statusCode)")
       guard response.statusCode < 300 else {
         let failureResponse = FailureResponse(responseCode: response.statusCode, errorData: data, headers: response.allHeaderFields)
+        Logger.shared.log(withLevel: .networking, message: "Error body: \(String(data: data, encoding: .utf8))")
         result(.failure(NetworkError.failureStatusCode(failureResponse: failureResponse)))
         return
       }
+      Logger.shared.log(withLevel: .networking, message: "Response headers: \(response.allHeaderFields.map{"\($0): \($1)"}.joined(separator: ", "))")
```
Comment (yafuquen) [inline TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift:44]: @SergioFierro printing response headers as it contains the request id

File: TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift [hunk 2]
```diff
@@ -37,53 +37,70 @@ class PushChallengeProcessor {
 
 extension PushChallengeProcessor: PushChallengeProcessorProtocol {
   func getChallenge(withSid sid: String, withFactor factor: PushFactor, success: @escaping ChallengeSuccessBlock, failure: @escaping TwilioVerifyErrorBlock) {
+    Logger.shared.log(withLevel: .info, message: "Getting challenge \(sid) with factor \(factor.sid)")
     challengeProvider.get(withSid: sid, withFactor: factor, success: success, failure: { error in
       failure(TwilioVerifyError.inputError(error: error as NSError))
     })
   }
   
   func updateChallenge(withSid sid: String, withFactor factor: PushFactor, status: ChallengeStatus, success: @escaping EmptySuccessBlock, failure: @escaping TwilioVerifyErrorBlock) {
```
Comment (AuthyApps) [inline TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift:46]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/db23e7504a6c641506761ed878f023492a1b05f0/TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift#L46"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 51 lines<br />
<code>function_body_length</code> <code>PushChallengeProcessor.swift:46</code></td>
    </tr>
  </tbody>
</table>

File: TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift [hunk 3]
```diff
@@ -43,6 +43,7 @@ extension PushFactory: PushFactoryProtocol {
   func createFactor(withAccessToken accessToken: String, friendlyName: String, pushToken: String, serviceSid: String,
```
Comment (AuthyApps) [inline TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift:43]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/db23e7504a6c641506761ed878f023492a1b05f0/TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift#L43"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 43 lines<br />
<code>function_body_length</code> <code>PushFactory.swift:43</code></td>
    </tr>
  </tbody>
</table>

File: TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift (hunk 1 above)
Comment (AuthyApps) [inline TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift:44]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/db23e7504a6c641506761ed878f023492a1b05f0/TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift#L44"></span>Opening braces should be preceded by a single space and on the same line as the declaration.<br />
<code>opening_brace</code> <code>URLSession+Extensions.swift:44</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.31%`**
TwilioVerify/TwilioVerify/Sources/Logger/DefaultLogger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.31%`**
TwilioVerify/TwilioVerify/Sources/Logger/DefaultLogger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.31%`**
TwilioVerify/TwilioVerify/Sources/Logger/DefaultLogger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`95.18%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.31%`**
TwilioVerify/TwilioVerify/Sources/Logger/DefaultLogger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**